frontend/src/components/mine/Permit/MinePermitTable.js

```diff
@@ -0,0 +1,467 @@
+
+    event.preventDefault();
+    props.openModal({
+      props: {
+        initialValues,
+        onSubmit: handleEditPermitAmendment,
+        title: `Edit permit amendment for ${permit.permit_no}`,
+      },
+      content: modalConfig.ADD_PERMIT_AMENDMENT,
+    });
+  };
+
+  // Add Permit Amendment
+  const handleAddPermitAmendment = (data) => {
+    props.createPermitAmendment(data.permit_guid, data).then(() => {
+      props.closeModal();
+    });
+  };
+  // Edit Permit
+  const handleEditPermit = (data) => {
+    props.updatePermit(data.permit_guid, data).then(() => {
+      props.closeModal();
+    });
+  };
+
+  // Add amalgamated Permit Amendment
+  const handleAddAlgPermitAmendment = (data) => {
+    const alg_data = { ...data, permit_amendment_type_code: "ALG" };
+    props.createPermitAmendment(data.permit_guid, alg_data).then(() => {
+      props.closeModal();
+    });
+  };
+
+  const openAddAmendmentModal = (event, onSubmit, title, permit_guid) => {
+    event.preventDefault();
+    props.openModal({
+      props: {
+        initialValues: { permit_guid },
+        onSubmit,
+        title,
+      },
+      content: modalConfig.ADD_PERMIT_AMENDMENT,
+    });
+  };
+
+  const openEditPermitModal = (event, permit_guid, permit_no) => {
+    const permit = props.permits.find((p) => p.permit_guid === permit_guid);
+
+    const initialValues = {
+      permit_status_code: permit.permit_status_code,
+      permit_guid,
+    };
+    event.preventDefault();
+    props.openModal({
+      props: {
+        initialValues,
+        onSubmit: handleEditPermit,
+        title: `Edit permit status for ${permit_no}`,
+      },
+      content: modalConfig.EDIT_PERMIT,
+    });
+  };
+
+  const openAddAmalgamatedPermitModal = (event, permit_guid, permit_no) =>
+    openAddAmendmentModal(
+      event,
+      handleAddAlgPermitAmendment,
+      `Add amalgamated permit to ${permit_no}`,
+      permit_guid
+    );
+
+  const openAddPermitAmendmentModal = (event, permit_guid, permit_no) =>
+    openAddAmendmentModal(
+      event,
+      handleAddPermitAmendment,
+      `Add permit amendment to ${permit_no}`,
+      permit_guid
+    );
+
+  const amendmentHistory = (record) => {
+    const childRowData = record.amendments.map((amendment, index) =>
+      transformChildRowData(
+        amendment,
+        record,
+        record.amendments.length - index,
+        props.major_mine_ind,
+        openEditAmendmentModal
+      )
+    );
+    return (
+      <Table
+        rowClassName={() => "permit-table-row"}
+        align="left"
+        pagination={false}
+        columns={childColumns}
+        dataSource={childRowData}
+      />
+    );
+  };
+
+  const rowData = props.permits.map((permit) =>
+    transformRowData(
+      permit,
+      props.partyRelationships,
+      props.major_mine_ind,
+      openEditPermitModal,
+      openAddPermitAmendmentModal,
+      openAddAmalgamatedPermitModal,
+      props.permitStatusOptions
+    )
+  );
```

Comment:
There are so many functions inside of this component. It looks like a lot of them are or could be pure functions that could be moved out of the exported function. This would improve readability, in my opinion.

Many of these actions (closing modals, sending API requests) look like things that should be passed in as props

Reply:
Let me make a refactoring pass as a bunch of piecemeal changes were made by different people. MinePermitInfo is the container component here so passing some of these in as props is the right approach.

Reply:
I committed a refactor that I think improves this. The open modal actions are passed in as props, and the handlers all exist on the PermitInfo level.